source/extensions/filters/network/thrift_proxy/binary_protocol_impl.cc

```diff
@@ -265,6 +265,106 @@ bool BinaryProtocolImpl::readBinary(Buffer::Instance& buffer, std::string& value
   return readString(buffer, value);
 }
 
+void BinaryProtocolImpl::writeMessageBegin(Buffer::Instance& buffer, const std::string& name,
+                                           MessageType msg_type, int32_t seq_id) {
+  BufferHelper::writeU16(buffer, Magic);
+  BufferHelper::writeU16(buffer, static_cast<uint16_t>(msg_type));
+  writeString(buffer, name);
+  BufferHelper::writeI32(buffer, seq_id);
+}
+
+void BinaryProtocolImpl::writeMessageEnd(Buffer::Instance& buffer) {
+  UNREFERENCED_PARAMETER(buffer);
+}
+
+void BinaryProtocolImpl::writeStructBegin(Buffer::Instance& buffer, const std::string& name) {
+  UNREFERENCED_PARAMETER(buffer);
+  UNREFERENCED_PARAMETER(name);
+}
+
+void BinaryProtocolImpl::writeStructEnd(Buffer::Instance& buffer) {
+  UNREFERENCED_PARAMETER(buffer);
+}
+
+void BinaryProtocolImpl::writeFieldBegin(Buffer::Instance& buffer, const std::string& name,
+                                         FieldType field_type, int16_t field_id) {
+  UNREFERENCED_PARAMETER(name);
+
+  BufferHelper::writeI8(buffer, static_cast<uint8_t>(field_type));
+  if (field_type == FieldType::Stop) {
+    return;
+  }
+
+  BufferHelper::writeI16(buffer, field_id);
+}
+
+void BinaryProtocolImpl::writeFieldEnd(Buffer::Instance& buffer) { UNREFERENCED_PARAMETER(buffer); }
+
+void BinaryProtocolImpl::writeMapBegin(Buffer::Instance& buffer, FieldType key_type,
+                                       FieldType value_type, uint32_t size) {
+  if (size > INT32_MAX) {
```

Comment:
nit: `std::numeric_limits` here and elsewhere?

Reply:
I'll do this in a follow-up.